pandas/core/resample.py

```diff
@@ -2173,7 +2173,11 @@ def _get_time_bins(self, ax: DatetimeIndex):
         if self.closed == "right":
             labels = binner
             if self.label == "right":
-                labels = labels[1:]
+                if binner.freqstr == "B":
+                    labels = labels[1:-1]
+                    bins = bins[0:-1]
+                else:
+                    labels = labels[1:]
```

Comment:
this isn't so clear to me - could you explain what happens here and how it fixes the issue please?